### PR TITLE
Enable jakartaee for session data

### PR DESCRIPTION
**Pull request by anjumfatima90**
for issue #12984

### CODE REVIEW

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionCache1.0.jakarta.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.sessionCache1.0.jakarta
+visibility = private
+
+-bundles= com.ibm.ws.session.jakarta, \
+  		  com.ibm.ws.session.cache.jakarta, \
+  		  com.ibm.ws.session.store.jakarta, \
+  		  com.ibm.websphere.javaee.jcache.1.1.jakarta; mavenCoordinates="javax.cache:cache-api:1.1.0"
```
Comment (tjwatson) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionCache1.0.jakarta.feature:9]: What is the deal with the maven coordinates.  Seems wrong to use javax here.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionDatabase1.0.jakarta.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.sessionDatabase1.0.jakarta
+visibility = private
+
+-bundles= com.ibm.ws.session.jakarta, \
+  		  com.ibm.ws.session.db.jakarta, \
+  		  com.ibm.ws.session.store.jakarta, \
+  		  com.ibm.websphere.javaee.jcache.1.1.jakarta; mavenCoordinates="javax.cache:cache-api:1.1.0"
```
Comment (tjwatson) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionDatabase1.0.jakarta.feature:9]: Same question about maven coordinates with javax.

File: dev/com.ibm.websphere.appserver.features/visibility/public/sessionDatabase-1.0/com.ibm.websphere.appserver.sessionDatabase-1.0.feature [hunk 3]
```diff
@@ -7,13 +7,10 @@ Manifest-Version: 1.0
...
- io.openliberty.servlet.api-3.0; ibm.tolerates:="3.1,4.0"; apiJar=false, \
- com.ibm.websphere.appserver.transaction-1.1; ibm.tolerates:=1.2, \
+ io.openliberty.servlet.api-3.0; ibm.tolerates:="3.1,4.0,5.0"; apiJar=false, \
+ com.ibm.websphere.appserver.transaction-1.1; ibm.tolerates:="1.2,2.0", \
  com.ibm.websphere.appserver.jdbc-4.0; ibm.tolerates:="4.1, 4.2, 4.3"
 -bundles=com.ibm.websphere.security, \
- com.ibm.ws.session.store, \
- com.ibm.ws.session.db, \
- com.ibm.ws.serialization, \
- com.ibm.ws.session
```
Comment (tjwatson) [inline dev/com.ibm.websphere.appserver.features/visibility/public/sessionDatabase-1.0/com.ibm.websphere.appserver.sessionDatabase-1.0.feature:17]: There was no `com.ibm.websphere.javaee.jcache.1.1` in the original `sessionDatabase-1.0` feature.  It looks like the jakarta version of it is not needed in the auto feature `io.openliberty.sessionDatabase1.0.jakarta`

File: dev/com.ibm.ws.session.cache_fat/bootstrapping.properties [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#Auto-generated properties by componenttest.common.apiservices.Bootstrap
+#Tue Sep 22 10:39:43 CDT 2020
+hostName=localhost
+junit_jar=../../ant_build/lib
+HOSTNAME.password=PASSWORD
+localhost.JavaHome=/Library/Java/JavaVirtualMachines/jdk1.8.0_172.jdk/Contents/Home/jre
```
Comment (tjwatson) [inline dev/com.ibm.ws.session.cache_fat/bootstrapping.properties:6]: This path has to be wrong for `localhost.JavaHome`.  Looks like it points to some Java install on your Mac.

File: dev/com.ibm.ws.session.cache_fat/bootstrapping.properties [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+local.sharedLib=../../test_prereqs/moonstone/sharedlibraries
+local.java=../../fattest.simplicity
+libertyInstallPath=../build.image/wlp
+shared.resources=../../build.sharedResources/lib
+HOSTNAME.user=USER
```
Comment (tjwatson) [inline dev/com.ibm.ws.session.cache_fat/bootstrapping.properties:11]: Is this file really used by anything?

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/config/HttpSession.java [hunk 6]
```diff
@@ -35,7 +35,7 @@
     private String cookieSameSite;
     private Integer maxInMemorySessionCount;
     private Boolean allowOverflow;
-    private Integer invalidationTimeout;
+    private String invalidationTimeout;
```
Comment (tjwatson) [inline dev/fattest.simplicity/src/com/ibm/websphere/simplicity/config/HttpSession.java:38]: Why did this change to String?

File: dev/com.ibm.ws.session.cache_fat/bootstrapping.properties (hunk 5 above)
Comment (anjumfatima90) [reply dev/com.ibm.ws.session.cache_fat/bootstrapping.properties:11]: I guess this is a locally generated file that got added. Will remove this file

File: dev/com.ibm.ws.session.cache_fat/bootstrapping.properties (hunk 4 above)
Comment (anjumfatima90) [reply dev/com.ibm.ws.session.cache_fat/bootstrapping.properties:6]: Will remove this file

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionCache1.0.jakarta.feature (hunk 1 above)
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionCache1.0.jakarta.feature:9]: This bundle for there in the original public feature. And javax.cache doesn't come under jakartaee 9.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionDatabase1.0.jakarta.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionDatabase1.0.jakarta.feature:9]: that got added by mistake. Will remove

File: dev/com.ibm.ws.session.cache_fat/bootstrapping.properties (hunk 4 above)
Comment (anjumfatima90) [reply dev/com.ibm.ws.session.cache_fat/bootstrapping.properties]: done!

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionDatabase1.0.jakarta.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.sessionDatabase1.0.jakarta.feature]: done!

**Comment by anjumfatima90** [comment]
#build

**Review by tjwatson — APPROVED** [review]

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#run-libby-bot

**Pull request merged**